### PR TITLE
stackrox: set e2e-benchmarking EXTRA_FLAGS to include --metrics-profile acs metrics url

### PR DESCRIPTION
**Pull request by davdhacs**
Test of https://github.com/kube-burner/kube-burner-ocp/pull/111

* add EXTRA_FLAGS parameter to perf&scale steps.
* for the stackrox perf&scale job(and future jobs), set EXTRA_FLAGS="--metrics-profile https://raw.githubusercontent.com/stackrox/stackrox/refs/heads/master/tests/performance/scale/tests/kube-burner/cluster-density/metrics.yml"

## New test-run with KUBE_BURNER_VERSION=1.4.0
- [x] Run with new release that includes the argument.

e2e-benchmarking log shows the $cmd used to call kube-burner-ocp includes the EXTRA_FLAGS string using the new kube-burner-ocp arg:
```
+ /tmp/kube-burner-ocp cluster-density-v2 --log-level=info --qps=20 --burst=20 --gc=true --uuid 8f61f145-274d-4532-910d-c313f1b0a6e1 --metrics-profile https://raw.githubusercontent.com/stackrox/stackrox/refs/heads/master/tests/performance/scale/tests/kube-burner/cluster-density/metrics.yml --gc-metrics=true --profile-type=both --iterations=216 --churn=true --es-server=https://XXXXXXXXX:XXXXXXXXXXXXXXXXXXXXXXXX@XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX --es-index=ripsaw-kube-burner
```

The kube-burner log then shows the url metrics file was loaded and the metrics targets inside that file are used:
```
time="2024-10-04 18:30:42" level=info msg="🔍 Endpoint: https://prometheus-k8s-openshift-monitoring.apps.ci-op-79p6y2l5-802fc.XXXXXXXXX.rox.systems; profile: config/https:/raw.githubusercontent.com/stackrox/stackrox/refs/heads/master/tests/performance/scale/tests/kube-burner/cluster-density/metrics.yml start: 2024-10-04T17:16:56Z end: 2024-10-04T18:25:48Z; job: cluster-density-v2" file="prometheus.go:69"
...
time="2024-10-04 18:31:12" level=info msg="Indexing [103574] documents from metric sensor_rox_sensor_events_network_policy_store_total" file="prometheus.go:233"
time="2024-10-04 18:31:29" level=info msg="Indexing finished in 16.859s: created=103574" file="prometheus.go:238"
```

### CODE REVIEW

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.16-nightly-x86.yaml [hunk 1]
```diff
@@ -55,7 +55,10 @@ tests:
     env:
       BASE_DOMAIN: perfscale.rox.systems
       COMPUTE_NODE_REPLICAS: "24"
+      E2E_REPOSITORY: https://github.com/davdhacs/e2e-benchmarking
```
Comment (rsevilla87) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.16-nightly-x86.yaml:58]: Just cut a new kube-burner-ocp version v1.4.0 including the patch for metrics-profiles, you can override the kube-burner version for this step with the env var KUBE_BURNER_VERSION https://github.com/cloud-bulldozer/e2e-benchmarking/blob/master/workloads/kube-burner-ocp-wrapper/run.sh#L11C23-L11C42

File: ci-operator/jobs/migtools/oadp-non-admin/migtools-oadp-non-admin-master-postsubmits.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ postsubmits:
     decoration_config:
       skip_cloning: true
     labels:
-      ci-operator.openshift.io/cluster: build11
+      ci-operator.openshift.io/cluster: build10
```
Comment (rsevilla87) [inline ci-operator/jobs/migtools/oadp-non-admin/migtools-oadp-non-admin-master-postsubmits.yaml:12]: Why are you updating this?

File: ci-operator/step-registry/openshift-qe/control-plane-smoke/openshift-qe-control-plane-smoke-ref.yaml [hunk 3]
```diff
@@ -14,6 +14,10 @@ ref:
     name: ocp-qe-gcp-sa
     mount_path: /ga-gsheet
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/control-plane-smoke/openshift-qe-control-plane-smoke-ref.yaml:17]: Not needed here

File: ci-operator/step-registry/openshift-qe/conc-builds/openshift-qe-conc-builds-ref.yaml [hunk 4]
```diff
@@ -16,6 +16,10 @@ ref:
       cpu: 100m
       memory: 100Mi
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/conc-builds/openshift-qe-conc-builds-ref.yaml:19]: Not needed here

File: ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml [hunk 5]
```diff
@@ -14,6 +14,10 @@ ref:
     name: perfscale-metal-bastion
     mount_path: /bm
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/ingress-perf/openshift-qe-ingress-perf-ref.yaml:17]: Not needed here

File: ci-operator/step-registry/openshift-qe/network-perf/openshift-qe-network-perf-ref.yaml [hunk 6]
```diff
@@ -6,6 +6,10 @@ ref:
     tag: latest
   cli: latest
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/network-perf/openshift-qe-network-perf-ref.yaml:9]: Not needed here

File: ci-operator/step-registry/openshift-qe/network-policy/openshift-qe-network-policy-ref.yaml [hunk 7]
```diff
@@ -8,6 +8,10 @@ ref:
   commands: openshift-qe-network-policy-commands.sh
   timeout: 8h0m0s
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/network-policy/openshift-qe-network-policy-ref.yaml:11]: Not needed here

File: ci-operator/step-registry/openshift-qe/network-smoke/openshift-qe-network-smoke-ref.yaml [hunk 8]
```diff
@@ -19,6 +19,10 @@ ref:
       memory: 100Mi
   timeout: 4h
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/network-smoke/openshift-qe-network-smoke-ref.yaml:22]: Not needed here

File: ci-operator/step-registry/openshift-qe/network-vm-perf/openshift-qe-network-vm-perf-ref.yaml [hunk 9]
```diff
@@ -6,6 +6,10 @@ ref:
     tag: latest
   cli: latest
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/network-vm-perf/openshift-qe-network-vm-perf-ref.yaml:9]: Not needed here

File: ci-operator/step-registry/openshift-qe/router-perf/openshift-qe-router-perf-ref.yaml [hunk 10]
```diff
@@ -14,6 +14,10 @@ ref:
     name: ocp-qe-gcp-sa
     mount_path: /ga-gsheet
   env:
+  - name: EXTRA_FLAGS
```
Comment (rsevilla87) [inline ci-operator/step-registry/openshift-qe/router-perf/openshift-qe-router-perf-ref.yaml:17]: Not needed here

**Comment by rsevilla87** [comment]
/pj-rehearse periodic-ci-openshift-qe-ocp-qe-perfscale-ci-main-aws-4.16-nightly-x86-control-plane-24nodes-acs

**Comment by rsevilla87** [comment]
Is this ready to merge @davdhacs ? if so, please add pj-rehearsal-ack

**Comment by jtaleric** [comment]
lgtm - only concern I would have is the number of metrics some of these queries return 😨 

```
time="2024-10-09 22:19:36" level=info msg="Indexing [113112] documents from metric central_rox_central_postgres_op_duration_bucket" file="prometheus.go:233"

```

**Comment by rsevilla87** [comment]
> > lgtm - only concern I would have is the number of metrics some of these queries return 😨
> > ```
> > time="2024-10-09 22:19:36" level=info msg="Indexing [113112] documents from metric central_rox_central_postgres_op_duration_bucket" file="prometheus.go:233"
> > ```
> 
> @mtodor should we change these metrics to reduce the volume before this runs (often)?

In the metrics file https://raw.githubusercontent.com/stackrox/stackrox/refs/heads/master/tests/performance/scale/config/metrics-acs.yml, you're capturing lot of raw metrics, you should consider adding some aggregation expressions (sum, rate, histogram_quantile, etc..) to reduce the number of documents

**Comment by jtaleric** [comment]
> @mtodor and I discussed the volume of these metrics and decided to start with this and iterate on aggregating and reducing the volume as we start using the data (and since we're in a separate elasticsearch, we will not be bad neighbor even if this is too much data right now)

ack!

**Pull request merged**